gcs/builder/builder.go

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) 2017 The btcsuite developers
+// Copyright (c) 2017 The Lightning Network Developers
+// Copyright (c) 2018 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package builder
```

Comment:
Any reason this is in a separate package and not just the `gcs` package?  It results in a generic and nondescriptive name as it is in a separate package.

It would be much clearer for the code to read like `gcs.BuildExtFilter(...)` instead of `builder.BuildExtFilter`.

Reply:
I agree with unifying the packages.  The builder package came from @Roasbeef's original implementation.  Unsure if there was a technical reason for the separation.

Reply:
There are many other functions exposed by the builder package that would need to be renamed if this change was made.  For example, `WithKeyPN` and the rest of the `With...` functions.  Currently these are called as `builder.With...` to return a `*builder.GCSBuilder`, but these function names wouldn't make much sense if they were qualified with the `gcs` name.

Reply:
After looking into this some more, it appears that the gcs package is intended to be a general implementation of golomb coded sets while builder is specific to building filters from decred blocks.  We can and probably also should remove the extra unused builder methods (everything but BuildBasicFilter and BuildExtFilter).